change: modify behavior of expressions, interaction between isBinary and override

0b5vr opened the pull request:
This PR updates the behavior of expressions.

- When an expression with isBinary overrides other expressions, it uses the binary output value to affect other expressions
- When an expression with isBinary is overridden by other expressions, the expression is completely suppressed if the effect received is greater than 0.0
- This PR also adds VRMExpression tests to confirm the desired behavior.

### Related link

- The original discussion: https://github.com/pixiv/three-vrm/issues/1484
- The spec change: vrm-c/vrm-specification#487

0b5vr committed:
change: modify behavior of expressions according the recent discussion

It's about interaction between isBinary and override expressions
See the discussion: https://github.com/pixiv/three-vrm/issues/1484

Also add a test of VRMExpression to confirm the desired behavior

0b5vr commented on packages/three-vrm-core/src/expressions/VRMExpression.ts:
```diff
@@ -1,6 +1,7 @@
 import * as THREE from 'three';
 import { VRMExpressionBind } from './VRMExpressionBind';
 import type { VRMExpressionOverrideType } from './VRMExpressionOverrideType';
+import type { VRMExpressionManager } from './VRMExpressionManager';
```
`VRMExpressionManager` is only referred from a doc comment

ke456-png commented on packages/three-vrm-core/src/expressions/VRMExpression.ts:
```diff
@@ -74,14 +79,25 @@ export class VRMExpression extends THREE.Object3D {
+   */
+  public get outputWeight(): number {
+    if (this.isBinary) {
+      return this.weight >= 0.5 ? 1.0 : 0.0;
```
```suggestion
      return this.weight > 0.5 ? 1.0 : 0.0;
```

> isBinary : A value greater than 0.5 is 1.0, otherwise 0.0
> https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm-1.0/expressions.md#expression-specification

ke456-png replied:
Please also modify the test cases.

0b5vr replied:
true! I misunderstood the spec.

0b5vr committed:
fix: expressions, update isBinary behavior according to the spec

I will update test cases in the next commit

See: https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm-1.0/expressions.md#expression-specification

The suggestion: https://github.com/pixiv/three-vrm/pull/1489

Co-authored-by: ke456-png <108649297+ke456-png@users.noreply.github.com>

0b5vr committed:
test: expressions, update test according to the spec

See: https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm-1.0/expressions.md#expression-specification

The suggestion: https://github.com/pixiv/three-vrm/pull/1489

0b5vr committed:
refactor: trivial unused code cleanup

ke456-png reviewed (approved):
NICE!
